### PR TITLE
MobIndexData changes

### PR DESCRIPTION
**Pull request by mattgodbolt**
* Made a few things ClassCase vs SHOUTY_OLD_STYLE
* Extracted a "Dice" class. Wrote tests.
* Extracted a few macros, get_hitroll and get_damroll
* Made test output a bit nicer (CatchFormatters.hpp)

### CODE REVIEW

**Comment by mattgodbolt** [comment]
Just here for show & tell. Will merge tomorrow if all still looks good (tested a bit locally, will test more).

**Comment by mattgodbolt** [comment]
Testing looks good: will fix up build errors.

File: src/MobIndexData.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    long vuln_flags{};
+    sh_int start_pos{};
+    sh_int default_pos{};
+    sh_int sex{};
+    sh_int race{};
+    long gold{};
+    long form{};
+    long parts{};
+    sh_int size{};
+    sh_int material{}; // TODO: is this actually used in any meaningful way?
```
Comment (snellers) [inline src/MobIndexData.hpp:49]: I'm fairly sure materials are used in things like calculations for breath attack, and the material vulnerabilities. Whether they look at this field specifically though I don't know without checking.

File: src/act_wiz.cpp [hunk 2]
```diff
@@ -1358,25 +1356,14 @@ void do_mfind(Char *ch, const char *argument) {
...
-    for (int vnum = 0; nMatch < top_mob_index; vnum++) {
-        if (auto *pMobIndex = get_mob_index(vnum)) {
-            nMatch++;
-            if (is_name(argument, pMobIndex->player_name)) {
-                found = true;
-                buffer += fmt::format("[{:5}] {}\n\r", pMobIndex->vnum, pMobIndex->short_descr);
-            }
-        }
-    }
+    auto buffer = ranges::accumulate(all_mob_indexes() | ranges::views::filter([&](const auto &mob) {
```
Comment (snellers) [inline src/act_wiz.cpp:1359]: nice

File: src/MobIndexData.hpp (hunk 1 above)
Comment (mattgodbolt) [reply src/MobIndexData.hpp:49]: I couldn't see the usage but I was also knee-deep in other stuff :)

File: src/act_wiz.cpp (hunk 2 above)
Comment (mattgodbolt) [reply src/act_wiz.cpp:1359]: Yeah, that was a bit of a flex :P

**Pull request merged**